src/macaron/slsa_analyzer/registry.py

```diff
@@ -69,11 +69,11 @@ def register(self, check: BaseCheck) -> None:
             logger.error("The check registered is invalid.")
             sys.exit(1)
 
-        # Update the check relationship mapping.
-        # If this check does not depend on any check,
-        # we add it as a standalone check.
+        # Update the check relationship mapping. If this check does not depend on any check,
+        # we add it as a standalone check ONLY when _check_relationships_mapping does not have an entry for it.
         if not check.depends_on:
-            self._check_relationships_mapping[check.check_id] = {}
+            if self._check_relationships_mapping.get(check.check_id, None) is None:
```

Comment:
Because `CheckResultType` [cannot](https://github.com/oracle-samples/macaron/blob/5615431c3d270c3fe1f40c82d1637637351806cd/src/macaron/slsa_analyzer/registry.py#L32) be `None`, to be more explicit it's better to directly check for inclusion here:
```suggestion
            if check.check_id not in self._check_relationships_mapping:
```

Reply:
Fixed in https://github.com/oracle-samples/macaron/pull/238/commits/eda34112639e7a762dffde66b18f3749217f6b83